linkerd/proxy/http/src/glue.rs

```diff
@@ -191,16 +193,16 @@ where
     }
 }
 
-impl<F> Future for HyperConnectFuture<F>
+impl<F, I, M> Future for HyperConnectFuture<F>
 where
-    F: TryFuture + 'static,
+    F: TryFuture<Ok = (I, M)> + 'static,
     F::Error: Into<Error>,
 {
-    type Output = Result<Connection<F::Ok>, Error>;
+    type Output = Result<Connection<I>>;
 
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         let this = self.project();
-        let transport = futures::ready!(this.inner.try_poll(cx)).map_err(Into::into)?;
+        let (transport, _meta) = futures::ready!(this.inner.try_poll(cx)).map_err(Into::into)?;
```

Comment:
nit, TIOLI: using a named ignored argument for the metadata (`_meta`) kind of implies to me that we want to keep it around until the end of the function even though we're not using it, but we _could_ drop it immediately by just writing
```suggestion
        let (transport, _) = futures::ready!(this.inner.try_poll(cx)).map_err(Into::into)?;
```

on the other hand, maybe it's nice to use the name to make it explicit _what_ we're ignoring here? idk, maybe disregard this one. :)